core/templates/domain/collection/collection-node.model.ts

```diff
@@ -76,7 +76,7 @@ export class CollectionNode {
   // Returns whether the exploration referenced by this node is private and
   // not published. This property is immutable. The value returned by this
   // function is undefined if doesExplorationExist() returns false.
-  isExplorationPrivate(): boolean {
+  isExplorationPrivate(): boolean | undefined {
```

Comment:
Do we actually need for this method to return `undefined`? Check the places where this method is used and if we somehow utilize null there. If the `undefined` is needed somewhere I would replace it with `null`.

Reply:
Okay I have changed the return type from undefined to null.  Originally the function definition returns boolean value if it executes succesfully else undefined! 
![image](https://user-images.githubusercontent.com/68547101/96958210-6dc6a500-151a-11eb-94a3-27d90bcd0db8.png)

Reply:
I know that but in the places where the method is used do we actually utilize this feature (that it sometimes returns `null`)?

Reply:
I did not find any place where we are currently using null return type .But I am not completely sure if removing null type will not cause any problem

Reply:
Also I guess we have a test case that directly inputs the function to be false.

Reply:
Yep, that is what I'm asking if it breaks anything or not. 

> Also I guess we have a test case that directly inputs the function to be false.

Not sure what you mean by this?

@nishantwrp Could you look into this? I did a quick look and think we can only return `boolean` here, but want the confirmation.